### PR TITLE
pkg/kubectl/bind: use native golang http server

### PR DESCRIPTION
**Pull request by s-urbaniak**

### CODE REVIEW

File: pkg/kubectl/bind/plugin/bind.go [hunk 1]
```diff
@@ -181,17 +168,22 @@ func (b *BindOptions) Run(ctx context.Context, urlCh chan<- string) error {
 			return err
 		}
 	}
+
+	auth := authenticator.NewLocalhostCallbackAuthenticator(10 * time.Minute)
+	err = auth.Start()
+	fmt.Fprintf(b.Options.ErrOut, "\n\n")
```
Comment (s-urbaniak) [inline pkg/kubectl/bind/plugin/bind.go:174]: why did we have this in the first place? (left unchanged for the refactor)

File: pkg/kubectl/bind/authenticator/authenticator.go [hunk 2]
```diff
@@ -44,113 +45,102 @@ func init() {
...
-	case <-ctx.Done():
-		return ctx.Err()
-	}
+	return nil
 }
 
-func (d *defaultAuthenticator) Endpoint(context.Context) string {
+// Endpoint returns the URL this server is listening to.
+// Start() must be called prior to this.
+func (d *LocalhostCallbackAuthenticator) Endpoint(context.Context) string {
```
Comment (sttts) [inline pkg/kubectl/bind/authenticator/authenticator.go:93]: why exactly the context? Is that some interface we have to follow?

File: pkg/kubectl/bind/plugin/bind.go (hunk 1 above)
Comment (sttts) [reply pkg/kubectl/bind/plugin/bind.go:174]: To separate it from other output I guess.

**Review by sttts — APPROVED** [review]

File: pkg/kubectl/bind/authenticator/authenticator.go (hunk 2 above)
Comment (s-urbaniak) [reply pkg/kubectl/bind/authenticator/authenticator.go:93]: leftover, it was unsued, let me remove

**Pull request merged**